quic-encryption-offload.md

```diff
@@ -199,27 +199,35 @@ Before the NDIS protocol driver posts any packets for a QEO connection, it first
 The `InformationBuffer` field of the `NDIS_OID_REQUEST` for this OID contains a pointer to an `NDIS_QUIC_CONNECTION`:
 
 ```C
-typedef enum {
-    Aes128Gcm,
-    Aes256Gcm,
-    ChaCha20Poly1305,
-    Aes128Ccm
+typedef enum _NDIS_QUIC_OPERATION_TYPE{
+    NDIS_QUIC_OPERATION_ADD,     // Add (or modify) a QUIC connection offload
+    NDIS_QUIC_OPERATION_REMOVE,  // Remove a QUIC connection offload
+} NDIS_QUIC_OPERATION_TYPE;
+
+typedef enum _NDIS_QUIC_DIRECTION_TYPE{
+    NDIS_QUIC_DIRECTION_TRANSMIT, // An offload for the transmit path
+    NDIS_QUIC_DIRECTION_RECEIVE,  // An offload for the receive path
+} NDIS_QUIC_DIRECTION_TYPE;
+
+typedef enum _NDIS_QUIC_CIPHER_TYPE{
+    NDIS_QUIC_CIPHER_TYPE_AEAD_AES_128_GCM,
+    NDIS_QUIC_CIPHER_TYPE_AEAD_AES_256_GCM,
+    NDIS_QUIC_CIPHER_TYPE_AEAD_CHACHA20_POLY1305,
+    NDIS_QUIC_CIPHER_TYPE_AEAD_AES_128_CCM,
 } NDIS_QUIC_CIPHER_TYPE;
 
 typedef struct _NDIS_QUIC_CONNECTION {
-    BOOLEAN IsAdd;
-    BOOLEAN IsTransmit;
-    NDIS_QUIC_CIPHER_TYPE CipherType;
-    uint8_t PayloadKeyLength;
-    uint8_t PayloadKey[32];
-    uint8_t HeaderKeyLength;
-    uint8_t HeaderKey[32];
-    uint8_t PayloadIv[12];
-    uint16_t Port; // Destination port.
+    NDIS_QUIC_OPERATION_TYPE Operation : 1;
+    NDIS_QUIC_DIRECTION_TYPE Direction : 1;
+    NDIS_QUIC_CIPHER_TYPE CipherType : 30;
     ADDRESS_FAMILY AddressFamily;
-    uint8_t Address[16]; // Destination IP address.
+    uint16_t UdpPort;         // Destination port.
     uint8_t ConnectionIdLength;
-    uint8_t ConnectionId[MAX_CID_LENGTH];
+    uint8_t Address[16];      // Destination IP address.
+    uint8_t ConnectionId[20]; // Limit to max of QUIC v1 & v2
```

Comment:
"Limit to max of QUIC v1 & v2"? I don't know what this is supposed to mean.

Reply:
QUIC versions 1 and 2 specify a max CID length of 20 bytes. Future versions may expand this further (up to 255), but there are no known plans to do so at this time. BTW, this was in the Winsock PR too.

Reply:
Main issue here is that it's unclear you are referring to the size.

Reply:
I'll take any recommended change you might suggest.

Reply:
QUIC v1 and v2 max CID size

Reply:
Done.